.github/sast_configurations/codeql.yml

```diff
@@ -0,0 +1,16 @@
+name: "CodeQL configurations"
+
+# initial configurations for CodeQL should be looking for high precision 
+# and high severity issues to ensure developers aren't compromised by false positives. 
+
+query-filters:
+  - include:
+      # alerts should only be raised for security issues to prevent being distracted with none security issues. 
+      tags contain: security
+      # High precision findings give the best likelihood that issues found are true positives.
+      precision:
+        - high
+        - very-high
+      # A security severity score of >=7 is classed as high or critical. This is an additional step to
+      # ensure that developers are not distracted by low severity, none issues. 
+      security-severity: /([7-9]|10)\.(\d)+/
```

Comment:
Does GitHub really not provide a less awful way to specify the minimum severity? This is pretty nasty and it's very hard to tell that it's correct without actually evaluating it.

Reply:
This is the only way I can see to filter the security severity level. The docs don't appear to provide a simpler method. 
The benefits of focusing on the high and critical issues first should mean that developers won't have so much of their time wasted dealing with false positives. The number of false positives in a project and therefore the value attained from this approach will vary from one to another. It is generally seen as good practice to start in this way and introduce the lower-severity findings further down the line. 

If all this was coming to was turning on CodeQL for GOV.UK repos, then I'd be inclined to agree with you that we should implement the default settings on each repo. Not because it's the right approach, but because the implementation is much simpler. However, down the road, we are going to have more SAST scans provided by SNYK, plus IaC scans, DAST scans, container scans, and possibly more. They will all require a process for building them into the CI pipelines of all the GOV.UK repos. It seems to me that a common approach should be taken for all these things. Then there are the additional processes around the scans. Somewhere the results can be stored, monitored, and analyzed together. This means the workflows will be extended down the line so that these features can be built in. An advanced implementation of CodeQL would be necessary at this point anyway as none of that can be done with a default implementation. We may as well take that approach now or we'll just have the default settings switched on for a while before we switch it off and reengineer it.

Reply:
These aren't filtering on the same criteria. Problem severity is looking at whether an alert is a warning, recommendation, or error. We want to look at security severity which is either critical, high, medium, or low. This is based on a numerical value which is why the regex is required.